tensorflow/python/framework/test_util.py

```diff
@@ -1639,10 +1639,25 @@ def use_deterministic_cudnn(func):
   def decorator(f):
 
     def decorated(self, *args, **kwargs):
-      original_var = os.environ.get("TF_CUDNN_DETERMINISTIC", "")
-      os.environ["TF_CUDNN_DETERMINISTIC"] = "true"
+      original_tf_cudnn_use_autotune = os.environ.get("TF_CUDNN_USE_AUTOTUNE")
+      os.environ["TF_CUDNN_USE_AUTOTUNE"] = "false"
+      original_xla_flags = os.environ.get("XLA_FLAGS")
+      new_xla_flags = "--xla_gpu_disable_autotune"
+      if original_xla_flags:
+        new_xla_flags += " " + original_xla_flags
```

Comment:
Assuming that the decorator takes precedence over XLA_FLAGS, new_xla_flags should be appended in stead of pre-pended. With the current change, if XLA_FLAGS contains --xla_gpu_disable_autotune=false, the decorator does not achieve its intended behavior.
If my assumption is incorrect, pre-pending is the wat to go

Reply:
Since this change was merged, I'll rebase [PR 35621](https://github.com/tensorflow/tensorflow/pull/35621)  and make the change.

Reply:
Well spotted and reported, @bas-aarts. I didn't see your comment in time to address it prior to the merge. Thank you for taking care of this in your PR.